src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -369,21 +383,22 @@ public Path getJarPath(String jarName) {
   /**
    * Checks whether the Cloud SDK path and version are valid.
    *
-   * @throws CloudSdkNotFoundException when an up-to-date Cloud SDK is not installed where expected
+   * @throws CloudSdkNotFoundException when Cloud SDK is not installed where expected
+   * @throws CloudSdkOutOfDateException when Cloud SDK is out of date
    */
-  public void validateCloudSdk() throws CloudSdkNotFoundException {
+  public void validateCloudSdk() throws CloudSdkNotFoundException, CloudSdkOutOfDateException {
     validateCloudSdkLocation();
     validateCloudSdkVersion();
   }
 
   private void validateCloudSdkVersion() {
     try {
       CloudSdkVersion version = getVersion();
-      if (version.getMajorVersion() < MINIMUM_VERSION) {
+      if (version.compareTo(MINIMUM_VERSION) < 0) {
         throw new CloudSdkOutOfDateException("Cloud SDK version " + version
-            + " is too old. Please update to at least " + MINIMUM_VERSION);
+            + " is too old. Please update to at least " + MINIMUM_VERSION, MINIMUM_VERSION);
       }
-    } catch (ProcessRunnerException | IllegalArgumentException ex) {
+    } catch (IllegalArgumentException | IOException ex) {
```

Comment:
perhaps we need to distinguish between the expected validation exception of being unsupported by the caller, and the exception of being unsupported by the core library.  And the messages should be built into the exception types with just the parameters passed into the constructor.